client/homebrew/editor/editor.jsx

```diff
@@ -189,7 +190,28 @@ const Editor = createClass({
 	brewJump : function(targetPage=this.getCurrentPage()){
 		if(!window || this.isMeta()) return;
 		// console.log(`Scroll to: p${targetPage}`);
-		window.frames['BrewRenderer'].contentDocument.getElementById(`p${targetPage}`).scrollIntoView({ behavior: 'smooth', block: 'start' });
+		const currentPos = window.frames['BrewRenderer'].contentDocument.getElementsByClassName('brewRenderer')[0].scrollTop;
+		const targetPos = window.frames['BrewRenderer'].contentDocument.getElementById(`p${targetPage}`).getBoundingClientRect().top;
+		const interimPos = targetPos >= 0 ? -30 : 30;
+
+		const bounceDelay = 100;
+		const scrollDelay = 500;
+
+		if(!this.state.isBrewScrolling) {
+			this.setState({
+				isBrewScrolling : true
+			});
```

Comment:
Do we need the state handling at all? I disabled this check and didn't notice any functional change.

If we don't need it, that would simplify this somewhat.

Reply:
It stops the function being spammed - without it, you can rapidly click the button to reverse up the page, before returning to the correct spot when you stop clicking it.

Reply:
Ah. In that case, it might be a little cleaner to use Lodash's `_.debounce()` function wrapper. See https://lodash.com/docs/4.17.15#debounce

Any function you wrap inside, like `_.debounce(jumpEditor, 150)` will only execute every 150 ms at most. Probably with the `'leading': true,` option so it triggers on the first click of a clicking spam, rather than the last click.

Reply:
Hmmm... I removed the state checks, wrapped the animation function in a `_.debounce` call... and now it's not executing anything in that block at all. If I recall correctly, it's also used in the editPage.jsx, I'm about to jump over there to check out a working example.

Reply:
I've mimicked the `editPage.jsx` usage of `_.debounce`, which has it functional again... however I'm now back at the original issue of needing a state to indicate that the animation is in progress, during which `debounce.cancel` should be called instead.

...I think I've got it now, using `throttle` instead of `debounce`. I'll push a commit shortly with the update.

Reply:
Awesome!

Reply:
There was some weird behavior, it looks better now with a slight change to the `throttle` options.

Reply:
Just fell down a rabbit hole; there are requests for a callback on completion of `scrollIntoView` going back at least six years; I'm pretty sure I saw some related issues that are closer to nine. It would make it a lot easier to lock out additional calls to the jump function if they'd been implemented.